cluster-autoscaler/core/scaleup/resource/manager_test.go

```diff
@@ -169,6 +170,38 @@ func TestApplyLimits(t *testing.T) {
 	}
 }
 
+type allocateResourcesTestCase struct {
+	resourcesLeft  Limits
+	resourcesDelta Delta
+	nodeCount      int
+	expectedOutput bool
+	expectedLeft   Limits
+}
+
+func TestAllocateResources(t *testing.T) {
+	testCases := []allocateResourcesTestCase{
```

Comment:
I would add a test case in which only one of the resources is exceeded, to emphasise that the method does not modify the resources left if it would result in exceeding the limit for one of them.

Reply:
Added.